scripts/pre_commit_linter.py

```diff
@@ -1165,19 +1168,422 @@ def __init__(self, all_filepaths, verbose_mode_enabled=False):
+            # the dir_patterns) is covered under CODEOWNERS.
+            for file_paths in _walk_with_gitignore('.', dir_patterns):
+                for file_path in file_paths:
+                    match = False
+                    for file_pattern in file_patterns:
+                        if file_path in glob.glob(file_pattern):
+                            match = True
+                            break
+                    if not match:
+                        print '%s is not covered under CODEOWNERS' % file_path
+                        failed = True
+
+            failed = failed or (
+                self.check_for_important_patterns_at_bottom_of_codeowners(
+                    important_rules_in_critical_section))
+
+            if failed:
+                summary_message = '%s   CODEOWNERS file check failed' % (
+                    _MESSAGE_TYPE_FAILED)
+            else:
+                summary_message = '%s  CODEOWNERS file check passed' % (
+                    _MESSAGE_TYPE_SUCCESS)
+
+            summary_messages.append(summary_message)
+            print summary_message
+            print ''
+
+        return summary_messages
+
+    def perform_all_lint_checks(self):
+        """Perform all the lint checks and returns the messages returned by all
+        the checks.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+        mandatory_patterns_messages = self._check_mandatory_patterns()
+        pattern_messages = self._check_bad_patterns()
+        codeowner_messages = self._check_codeowner_file()
+        return (
+            mandatory_patterns_messages + pattern_messages + codeowner_messages)
+
+
+class JsTsLintChecksManager(LintChecksManager):
+    """Manages all the Js and Ts linting functions.
+
+    Attributes:
+        all_filepaths: list(str). The list of filepaths to be linted.
+        js_filepaths: list(str): The list of js filepaths to be linted.
+        ts_filepaths: list(str): The list of ts filepaths to be linted.
+        parsed_js_and_ts_files: dict. Contains the content of JS files, after
+            validating and parsing the files.
+        verbose_mode_enabled: bool. True if verbose mode is enabled.
+    """
+    def __init__(self, js_filepaths, ts_filepaths, verbose_mode_enabled=False):
+        """Constructs a JsTsLintChecksManager object.
+
+        Args:
+            js_filepaths: list(str). The list of js filepaths to be linted.
+            ts_filepaths: list(str). The list of ts filepaths to be linted.
+            verbose_mode_enabled: bool. True if verbose mode is enabled.
+        """
+        node_path = os.path.join(os.pardir, 'oppia_tools/node-10.15.3')
+        os.environ['PATH'] = '%s/bin:' % node_path + os.environ['PATH']
+
+        self.all_filepaths = js_filepaths + ts_filepaths
+        self.js_filepaths = js_filepaths
+        self.ts_filepaths = ts_filepaths
+        LintChecksManager.__init__(
```

Comment:
I believe calling it the parent class like this might cause some issues, use [super](https://docs.python.org/2/library/functions.html#super) instead.

Reply:
Updated in the new commit